suisa_sendemeldung/suisa_sendemeldung.py

```diff
@@ -404,14 +454,23 @@ def create_message(sender, recipient, subject, text, filename, csv, cc=None, bcc
     msg["Subject"] = subject
     # set body
     msg.attach(MIMEText(text))
-    # attach csv
-    part = MIMEBase("text", "csv")
-    part.set_payload(csv.encode("utf-8"))
-    encode_base64(part)
-    part.add_header(
-        "Content-Disposition", f'attachment; filename="{basename(filename)}'
-    )
-    msg.attach(part)
+    if filetype == "xlsx":
+        part = MIMEBase("application", "vnd.ms-excel")
+        part.set_payload(data.getvalue())
+        encode_base64(part)
+        part.add_header(
+            "Content-Disposition", f'attachment; filename="{basename(filename)}'
+        )
+        msg.attach(part)
+    elif filetype == "csv":
+        # attach csv
+        part = MIMEBase("text", "csv")
+        part.set_payload(data.encode("utf-8"))
+        encode_base64(part)
+        part.add_header(
+            "Content-Disposition", f'attachment; filename="{basename(filename)}'
+        )
+        msg.attach(part)
```

Comment:
i keep wanting to simplify this, maybe it should move to it's own function, the following doesn't improve on it
```suggestion
    part = None
    if filetype == "xlsx":
        part = MIMEBase("application", "vnd.ms-excel")
        part.set_payload(data.getvalue())
        encode_base64(part)
    elif filetype == "csv":
        # attach csv
        part = MIMEBase("text", "csv")
        part.set_payload(data.encode("utf-8"))
        encode_base64(part)
    if part:
        part.add_header(
            "Content-Disposition", f'attachment; filename="{basename(filename)}'
        )
        msg.attach(part)
```